client/lomas_client/libraries/opendp.py

```diff
@@ -0,0 +1,117 @@
+from typing import Optional, Type
+
+import opendp as dp
+from fastapi import status
+from lomas_core.models.requests import (
+    OpenDPDummyQueryModel,
+    OpenDPQueryModel,
+    OpenDPRequestModel,
+)
+from lomas_core.models.responses import CostResponse, QueryResponse
+
+from lomas_client.constants import DUMMY_NB_ROWS, DUMMY_SEED
+from lomas_client.http_client import LomasHttpClient
+from lomas_client.utils import raise_error
+
+
+class OpenDPClient:
+    """A client for executing and estimating the cost of OpenDP queries."""
+
+    def __init__(self, http_client: LomasHttpClient):
+        self.http_client = http_client
+
+    def cost(
+        self,
+        opendp_pipeline: dp.Measurement,
+        fixed_delta: Optional[float] = None,
+    ) -> Optional[CostResponse]:
+        """This function estimates the cost of executing an OpenDP query.
+
+        Args:
+            opendp_pipeline (dp.Measurement): The OpenDP pipeline for the query.
+            fixed_delta (Optional[float], optional): If the pipeline measurement is of\
+                type “ZeroConcentratedDivergence” (e.g. with make_gaussian) then it is\
+                converted to “SmoothedMaxDivergence” with make_zCDP_to_approxDP\
+                (`See Smartnoise-SQL postprocessing documentation.\
+                <https://docs.smartnoise.org/sql/advanced.html#postprocess>`__).\
+                In that case a fixed_delta must be provided by the user.\
+                Defaults to None.
+
+        Returns:
+            Optional[dict[str, float]]: A dictionary containing the estimated cost.
+        """
+        opendp_json = opendp_pipeline.to_json()
+        body_dict = {
+            "dataset_name": self.http_client.dataset_name,
+            "opendp_json": opendp_json,
+            "fixed_delta": fixed_delta,
+        }
+
+        body = OpenDPRequestModel.model_validate(body_dict)
+        res = self.http_client.post("estimate_opendp_cost", body)
+
+        if res.status_code == status.HTTP_200_OK:
+            data = res.content.decode("utf8")
+            return CostResponse.model_validate_json(data)
+
+        raise_error(res)
```

Comment:
I wonder if these lines could be in a utils function as it is the same 4 times (4 libraries queries) (?)

Reply:
Good point, will do!